source/loader/layers/tracing/ur_tracing_layer.cpp

```diff
@@ -28,13 +28,14 @@ context_t::context_t() {
     xptiFrameworkInitialize();
 
     call_stream_id = xptiRegisterStream(CALL_STREAM_NAME);
+    traceEnvEnabled = getenv_tobool("UR_ENABLE_TRACING");
```

Comment:
This is a rock-and-a-hard-place type situation. There is already an XPTI env var that you need to set to enable the tracing used by this layer (actually there are several) - adding this new env var is our rock (detracts from user experience, seems generally redundant).
So the obvious thing would be to treat the XPTI one as our layer-enabling env var instead of adding the new one - but if we do that the API mechanism becomes redundant. Enabling the XPTI variable is always needed for the layer to work - you'd need to set it even if you wanted to enable your layers via api. This is our hard place, this solution makes the API redundant, and might lead to users making unnecessary API calls. At best it would be weird to document.
There is also a third, un-named, unpleasant location: we could just not report this tracing layer through the api mechanism and leave it as a kind of power-user option. This would preserve the current behaviour for existing users without being potentially misleading by reporting its existence through the loader config mechanism. That kind of sucks though because we have this functionality so we should encourage people to use it. It would also feel weird to have a layer that's fundamentally inconsistent with the layer mechanism we're trying to set up.

Reply:
I had the same dilemma. I imagine most users will leverage this layer indirectly through tools like `urtrace` or just `sycl`, so the ergonomics of this are not that important. It's already complicated and requires at least some understanding of the XPTI framework.
So I'm fine with adding the new env variable, provided it's well documented.

Reply:
Cool, added some documentation of the new env var mechanism and all the layers in INTRO.rst